AdyenActions/UI/View Controllers/QR Code/QRCodeView.swift

```diff
@@ -0,0 +1,204 @@
+        logo.heightAnchor.constraint(equalToConstant: logoSize.height).isActive = true
+        logo.accessibilityIdentifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "logo")
+        
+        logo.imageURL = model.logoUrl
+        
+        return logo
+    }()
+    
+    private lazy var instructionLabel: UILabel = {
+        let instructionLabel = UILabel(style: model.style.instructionLabel)
+        instructionLabel.text = model.instruction
+        instructionLabel.numberOfLines = 0
+        instructionLabel.accessibilityIdentifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "instructionLabel")
+        
+        return instructionLabel
+    }()
+    
+    private lazy var qrCodeImageView: UIImageView = {
+        let data = model.action.qrCodeData.data(using: String.Encoding.ascii)
+        if let filter = CIFilter(name: "CIQRCodeGenerator") {
+            filter.setValue(data, forKey: "inputMessage")
+            let transform = CGAffineTransform(scaleX: 3, y: 3)
+
+            if let output = filter.outputImage?.transformed(by: transform) {
+                return UIImageView(image: UIImage(ciImage: output))
+            }
+        }
+        return UIImageView()
+    }()
+    
+    private lazy var amountToPayLabel: UILabel = {
+        let amountToPayLabel = UILabel(style: model.style.amountToPayLabel)
+        amountToPayLabel.numberOfLines = 0
+        if let currencyCode = model.payment?.amount.currencyCode {
+            amountToPayLabel.text = model.payment?.amount.formatted
+        }
+        amountToPayLabel.accessibilityIdentifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "amountToPayLabel")
+        
+        return amountToPayLabel
+    }()
+    
+    private lazy var progressView: UIProgressView = {
+        let progressViewSize = CGSize(width: 120, height: 4)
+        let progressView = UIProgressView(style: model.style.progressView)
+        progressView.observedProgress = model.observedProgress
+        progressView.widthAnchor.constraint(equalToConstant: progressViewSize.width).isActive = true
+        progressView.heightAnchor.constraint(equalToConstant: progressViewSize.height).isActive = true
+        progressView.accessibilityIdentifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "progressView")
+        
+        return progressView
+    }()
+    
+    private lazy var expirationLabel: UILabel = {
+        let expirationLabel = UILabel(style: model.style.expirationLabel)
+        expirationLabel.numberOfLines = 1
+        expirationLabel.accessibilityIdentifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "expirationLabel")
+        
+        bind(model.expiration, to: expirationLabel, at: \.text)
+        
+        return expirationLabel
+    }()
+
+    @objc private func saveQRCodeAsImage() {
+        delegate?.saveAsImage(qrCodeImage: qrCodeImageView.adyen.snapShot(), sourceView: saveAsImageButton)
+    }
```

Comment:
most of this code is very similar to `PixView`, this seems like code duplication to me :-)